# HG changeset patch
# User Serviscope Minor <serviscope_minor@verybigfrog.com>
# Date 1407508549 -3600
#      Fri Aug 08 15:35:49 2014 +0100
# Node ID 2ceb734a663f9a1bdcb69219378cfac714532aea
# Parent  956fc864c39f5fbdd83f7773b1daa8e898a726f3
Set axis label colors correctly when plotting with gnuplot.

* __go_draw_axes__.m: Change the xlabel colour from gca.xcolor to
gca.xlabel.color.  Same for y and z labels.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -160,17 +160,17 @@ function __go_draw_axes__ (h, plot_strea
                undo_string_escapes (tt), fontspec, colorspec, 
                __do_enhanced_option__ (enhanced, t));
     endif
   endif
 
   if (! isempty (axis_obj.xlabel))
     t = get (axis_obj.xlabel);
     angle = t.rotation;
-    colorspec = get_text_colorspec (axis_obj.xcolor, mono);
+    colorspec = get_text_colorspec (t.color, mono);
     if (isempty (t.string))
       fprintf (plot_stream, "unset xlabel;\n");
       fprintf (plot_stream, "unset x2label;\n");
     else
       [tt, f, s] = __maybe_munge_text__ (enhanced, t, "string");
       fontspec = create_fontspec (f, s, gnuplot_term);
       if (strcmpi (axis_obj.xaxislocation, "top"))
         fprintf (plot_stream, "set x2label \"%s\" %s %s %s",
@@ -188,17 +188,17 @@ function __go_draw_axes__ (h, plot_strea
         fprintf (plot_stream, "unset x2label;\n");
       endif
     endif
   endif
 
   if (! isempty (axis_obj.ylabel))
     t = get (axis_obj.ylabel);
     angle = t.rotation;
-    colorspec = get_text_colorspec (axis_obj.ycolor, mono);
+    colorspec = get_text_colorspec (t.color, mono);
     if (isempty (t.string))
       fprintf (plot_stream, "unset ylabel;\n");
       fprintf (plot_stream, "unset y2label;\n");
     else
       [tt, f, s] = __maybe_munge_text__ (enhanced, t, "string");
       fontspec = create_fontspec (f, s, gnuplot_term);
       if (strcmpi (axis_obj.yaxislocation, "right"))
         fprintf (plot_stream, "set y2label \"%s\" %s %s %s",
@@ -216,17 +216,17 @@ function __go_draw_axes__ (h, plot_strea
         fprintf (plot_stream, "unset y2label;\n");
       endif
     endif
   endif
 
   if (! isempty (axis_obj.zlabel))
     t = get (axis_obj.zlabel);
     angle = t.rotation;
-    colorspec = get_text_colorspec (axis_obj.zcolor, mono);
+    colorspec = get_text_colorspec (t.color, mono);
     if (isempty (t.string))
       fputs (plot_stream, "unset zlabel;\n");
     else
       [tt, f, s] = __maybe_munge_text__ (enhanced, t, "string");
       fontspec = create_fontspec (f, s, gnuplot_term);
       fprintf (plot_stream, "set zlabel \"%s\" %s %s %s",
                undo_string_escapes (tt), colorspec, fontspec,
                __do_enhanced_option__ (enhanced, t));
